src/Pos/Worker.hs

```diff
@@ -58,6 +60,12 @@ allWorkers NodeResources {..} = mconcatPair
     , wrap' "delegation" $ dlgWorkers
     , wrap' "slotting"   $ (properSlottingWorkers, mempty)
 
+    , case ncTopology ncNetworkConfig of
+        TopologyBehindNAT dnsDomains ->
+          wrap' "subscription" $ subscriptionWorkers ncNetworkConfig dnsDomains
+        _otherwise ->
+          mempty -- No subscription worker required
```

Comment:
Ah right, you have to actually give a kademlia yaml file, else it won't go. Should throw an informative exception there I suppose.